packages/xml-views-validation/src/validators/attributes/unknown-attribute-key.ts

```diff
@@ -0,0 +1,179 @@
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (attribute.key === "binding") {
+    return [];
+  }
+
+  // Special case: "stashed" property can appear under sap.uxap.ObjectPageLazyLoader
+  // Technically it can be under any Element but it should only be used for this one.
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (
+    attribute.key === "stashed" &&
+    ui5NodeToFQN(ui5Class) === "sap.uxap.ObjectPageLazyLoader"
+  ) {
+    return [];
+  }
+
+  const allProps: (
+    | UI5Prop
+    | UI5Event
+    | UI5Association
+    | UI5Aggregation
+  )[] = flattenProperties(ui5Class);
+  const allEvents = flattenEvents(ui5Class);
+  const allAssociations = flattenAssociations(ui5Class);
+  // Aggregations can be used as attributes for binding
+  const allAggregations = flattenAggregations(ui5Class);
+  const allClassUI5Attributes = allProps
+    .concat(allEvents)
+    .concat(allAssociations)
+    .concat(allAggregations);
+  // TODO Should we filter out non-public/protected names?
+  if (
+    find(allClassUI5Attributes, (_) => _.name === attribute.key) !== undefined
+  ) {
+    return [];
+  }
+
+  return [
+    {
+      kind: "UnknownAttributeKey",
+      message: `Unknown attribute key: ${attribute.key}`,
+      offsetRange: {
+        start: attribute.syntax.key.startOffset,
+        end: attribute.syntax.key.endOffset,
+      },
+      severity: "error",
+    },
+  ];
+}
+
+function isAttributeNameAlwaysValid(attribute: XMLAttribute): boolean {
+  // This will never happen - it was checked before calling this function
+  /* istanbul ignore if */
+  if (attribute.key === null) {
+    return true;
+  }
+  // xmlns attributes are always valid
+  // TODO "xmlns:" should not return true (change when using new function in xml-utils)
+  if (isXMLNamespaceKey(attribute.key)) {
+    return true;
+  }
+  const { ns, name } = splitAttributeByNamespace(attribute);
+  // sap.ui.dt:designtime is always valid
+  if (ns === "sap.ui.dt" && name === "designtime") {
+    return true;
+  }
+  // sap.ui.core:require is always valid
+  // TODO this is not true for FragmentDefinition inside xml views
+  // See: https://openui5nightly.hana.ondemand.com/topic/b11d853a8e784db6b2d210ef57b0f7d7
+  if (ns === "sap.ui.core" && name === "require") {
+    return true;
+  }
+
+  // templates require is always valid
+  if (ns === TEMPLATING_NS && name === "require") {
+    return true;
+  }
+
+  // All attributes in the custom data namespace are always valid
+  if (ns === CUSTOM_DATA_NS) {
+    return true;
+  }
+
+  return false;
+}
+
+function splitAttributeByNamespace(
+  attribute: XMLAttribute
+): { ns: string | undefined; name: string | undefined } {
+  // This will never happen - it was checked before calling this function
+  /* istanbul ignore if */
+  if (attribute.key === null) {
```

Comment:
may be able to model this assumption at the signature level:

XMLAttribute & {key:string}